repos/spack_repo/builtin/packages/erf/package.py

```diff
@@ -4,6 +4,15 @@
 from spack.package import *
 
 
+def submodules(package):
+    submodules = []
+
+    with when("+noahmp"):
+        submodules.append("Submodules/Noah-MP")
```

Comment:
`when` blocks don't work this way for imperative code, only for the declarative directives (things that take a `when` argument).

```suggestion
    if package.spec.satisfies("+noahmp"):
        submodules.append("Submodules/Noah-MP")
```

Reply:
I'm working on a package for NoahMP as well; however, development on that is going to be delayed because of a lack of funding. So once I've finished that I intend to swap it out for the Spack package.